core/src/main/scala/better/files/ThreadBackedFileMonitor.scala

```diff
@@ -60,15 +57,12 @@ abstract class ThreadBackedFileMonitor(val root: File, maxDepth: Int) extends Fi
     }
   }
 
-  override def start() = {
+  override def start()(implicit executionContext: ExecutionContext) = {
     watch(root, maxDepth)
-    thread.start()
+    executionContext.execute(() => Iterator.continually(service.take()).foreach(process))
```

Comment:
Won't this submit just one task to the executor? `take` will block and `process` will run on the same thread no? If `process` is expensive, it can't `take` another until completion, no? Is that the design intent?